tools/lint/lint.py

```diff
@@ -624,6 +624,32 @@ def check_script_metadata(repo_root, path, f):
     return errors
 
 
+def check_multiple_identical_values(repo_root, paths):
+    """
+    Runs lints that check for multiple identical paths
+    in all_paths_lints.
+
+    :param repo_root: the repository root
+    :param paths: a list of all the paths within the repository
+    :returns: a list of errors found in ``path``
+    """
+
+    file_dict = dict()
+    errors = []
+    for paths in paths:
+        file, file_extension =  os.path.splitext(paths)
+        if file not in file_dict.keys():
+            file_dict[file] = list()
+            file_dict[file].append(file_extension)
+        else:
+            file_dict[file].append(file_extension)
+    for f in file_dict.keys():
+        if len(file_dict[f]) > 0:
+            errors.extend(paths(repo_root, paths))
```

Comment:
So the errors list should be a list of tuples of the form `(type, error, path, line)`. In these cases error should be some code like `DUPLICATE EXTENSIONLESS PATH`, `error` is a sitring like `Got files for path %s with extensions %s, paths exclusing extensions must be unique` with the `%s` values fileld in from the data, `path` can be the path without an extension, and `line` will be `None`.